assets/src/apps/authoring/components/PropertyEditor/custom/CompJsonEditor.tsx

```diff
@@ -0,0 +1,77 @@
+import { selectCurrentSelection } from 'apps/authoring/store/parts/slice';
+import React, { CSSProperties, Fragment, useState } from 'react';
+import { useEffect } from 'react';
+import { Button, Modal } from 'react-bootstrap';
+import { useSelector } from 'react-redux';
+interface JsonEditorProps {
+  jsonValue: any;
+  onChange: (changedJson: any) => void;
+  schema: any;
+  existingPartIds: string[];
+}
+const CompJsonEditor: React.FC<JsonEditorProps> = (props) => {
+  const { jsonValue, onChange, existingPartIds, schema } = props;
+  const val = { id: jsonValue.id, custom: jsonValue.custom };
+  const [value, setValue] = useState<string>(JSON.stringify(val, null, 4));
+  const [validationMsg, setValidationMsg] = useState<string>('');
+  const currentPartSelection = useSelector(selectCurrentSelection);
+  const [displayEditor, setDisplayEditor] = useState<boolean>(false);
+  const textAreaStyle: CSSProperties = {
+    width: '100%',
+  };
+  useEffect(() => {
+    try {
+      const jsonVal = JSON.parse(value);
+      if (existingPartIds.indexOf(jsonVal.id) !== -1 && currentPartSelection !== jsonVal.id) {
```

Comment:
if we have `currentPartSelection !== jsonVal.id` doesn't it mean that they will be forced to change the ID?

Reply:
existingIds will also contain the currentPartSelection.... so just to restrict the duplicate ID validation this extra condition is there.